mxcube3/core/limsutils.py

```diff
@@ -448,6 +450,11 @@ def synch_with_lims():
         sample_info["defaultPrefix"] = get_default_prefix(sample_info, False)
         sample_info["defaultSubDir"] = get_default_subdir(sample_info)
 
+        regexp = re.compile("^[a-zA-Z0-9:+_-]+$")
+
+        if not regexp.match(sample_info["sampleName"]):
+            raise AttributeError("sample name contains an incorrect character")
```

Comment:
Suggestion, define a global variable out-side of this function, e.g:

VALID_SAMPLE_NAME_REGEXP = re.compile("^[a-zA-Z0-9:+_-]+$")

and then use it on line 455 as :

if not VALID_SAMPLE_NAME_REGEXP.match(sample_info["sampleName"]):

This way you compile the regexp only once, when the module is loaded. Otherwise  there is no point using compiled regexp here, it is compiled each time we enter the function.